src/main/java/baritone/pathing/movement/movements/MovementParkour.java

```diff
@@ -265,7 +276,12 @@ public MovementState updateState(MovementState state) {
             }
         } else if (!ctx.playerFeet().equals(src)) {
             if (ctx.playerFeet().equals(src.offset(direction)) || ctx.player().posY - src.y > 0.0001) {
-                if (!MovementHelper.canWalkOn(ctx, dest.down()) && !ctx.player().onGround && MovementHelper.attemptToPlaceABlock(state, baritone, dest.down(), true, false) == PlaceResult.READY_TO_PLACE) {
+                if (Baritone.settings().allowPlace.value
```

Comment:
I don't know about this addition - if `allowPlace` is false we need to bail out MUCH earlier than here. At this point we're **in midair** and trying to parkour place. It would be unsafe and could kill the player to attempt a parkour place but then refuse to actually place it because of this setting having the wrong value.

Reply:
We don't attempt to parkour place when `allowPlace` is false. This new check is there to reject placing when the block we are going to land on is not yet turned into water (but frostwalker can fail so I can't just say this is not a problem).

A straight traverse-traverse-descend-parkour chain has a 0.7% failure rate when checking `allowPlace` and a 19.9% chance of doing a forbidden placement without checking `allowPlace`. Doing straight descend-traverse-traverse-parkour instead yields 0.2% failure rate and 19.4% forbidden placements instead (test sizes are 1000-1200 runs each and I also have 50-70 runs each on recording if you want to see them, they are just too big to put them into this comment right away)

If you give me a nice name for it (maybe one fitting for #1586?) I will look into allowing it to place even with `allowPlace` disabled (delayed to reduce false positives).

Reply:
What's an example of such a forbidden placement? Like, when you're doing a parkour jump to ice, but the ice melts while you're in midair? How exactly does this happen?

Reply:
I can't visualize what you mean by `This new check is there to reject placing when the block we are going to land on is not yet turned into water`

Reply:
You are @leijurv:leijurv.com on matrix right? Can I just message you an 105MB zip file (all four) or a 40MB mp4 (only first)?

Reply:
Forbidden placement it just placing even though it is neither planned nor allowed.

Reply:
Yeah, that's cool

Reply:
Although it might need to be in separate files since I have a vague memory that my shitty nginx has a 50mb upload limit lol

Reply:
Doesn't seem like I can message you. I can't even message myself on a different homeserver.

Reply:
Ok. Feel free to share the mp4 by any other means - youtube, dropbox, google drive, wetransfer.com. Actually now I remember you refuse to use discord so maybe you are one of that kind 😉 of person 😉, you might want a FOSS solution? Maybe try just an account on the matrix.org homeserver using app.element.io? I have one of those, it's `@leijurv:matrix.org`. Might work?

Reply:
Also maybe try again now, I just updated my synapse. I bet it was YEARS out of date I literally can't remember the last time I messed with it. Probably not updated since late 2021 lol

Reply:
i cant see any messages from you in any of the three rooms :(

Reply:
3? Only saw two so far.
I can see your messages but mine just say "didn't send"

Reply:
Found the third one but:
![image](https://user-images.githubusercontent.com/68165024/212196921-41485685-abd2-4853-8dd3-736957f8386a.png)

Reply:
Hm, even the most recent where it's just both of us `@matrix.org`? cringe foss software. propietary software is superior lol

Reply:
maybe can you just find some place to upload the file that you're comfortable with, then paste a link here?

Reply:
Fallback would have been thrashing my uploads repo (which is probably already bigger than baritone)

Reply:
Of what (or whom?) exactly are you making fun here?

Reply:
Whoops. Just reread this conversation noticed that `This new check is there to reject placing when the block we are going to land on is not yet turned into water` should have been `This new check is there to reject placing when the block we are going to land on is not yet turned into ice` instead. Might have caused a good part of the conversation above.